### ISSUE TITLE
Execution chain is broken by sanboxify

### ISSUE THREAD

**Issue by ceache** [cloudflare/sandbox#7]
This is easily triggered by trying to check the return value:

``` shell
$ SECCOMP_SYSCALL_DENY="" ./sandboxify /bin/false || echo boo
$ SECCOMP_SYSCALL_DENY="" ./sandboxify /bin/false && echo boo
boo
```
The return value is always 0, taken for the `EXIT_SUCCESS` of `sanboxify`, not the actually executed binary.

Similarly, the execution chain amounts to a double fork and exec, rendering supervision difficult.
``` shell
$ SECCOMP_SYSCALL_DENY="" ./sandboxify /bin/sleep inf
initializing seccomp with default action (allow)
$ # we get prompt back??
```
`sleep` is now daemonized and re-parented to init (or whatever is the top level process reaper)

The solution to both issues would be to reverse the child / parent roles and instead have the child trace the parent and the parent exec into the final binary.
If interested, I have a rough PR that does just that.

**Comment by ignatk** [comment]
> sleep is now daemonized and re-parented to init (or whatever is the top level process reaper)

It is assumed that most of the time `sandboxify` will be launched from a service manager, like systemd (which would be just a "forking" service in systemd terminology), so re-parenting the sandbox process is considered an OK/expected behaviour.

But if we can improve the cases of launching `sandboxify` from the shell and make it similar to the way it works with `libsandbox.so` (the sandboxed process is a direct child of the shell) - that would be great!

**Comment by ignatk** [comment]
Fixed in #8
